src/scale.js

```diff
@@ -29,8 +30,9 @@ function scaleLimitDefined(scaleOptions, limit, suggestedLimit) {
 
 function verifyScaleIDs(annotation, scales) {
   for (const key of ['scaleID', 'xScaleID', 'yScaleID']) {
-    if (annotation[key] && !scales[annotation[key]] && verifyProperties(annotation, key)) {
-      console.warn(`No scale found with id '${annotation[key]}' for annotation '${annotation.id}'`);
+    const scaleID = retrieveScaleID(scales, annotation, key);
+    if (scaleID && !scales[scaleID] && verifyProperties(annotation, key)) {
```

Comment:
Can `scaleID` still be undefined?

Reply:
uhm... you are right. it's not possible. The first check must be removed.

Reply:
I checked better and the `retrieveScaleID` function can return an undefined value when the `scaleID` options is not set. See above comment related to `scaleID` option.